### ISSUE TITLE
Loop with dict insertions refactors to invalid code

### ISSUE THREAD

**Issue by marcoaaguiar** [sourcery-ai/sourcery#85]
### Issue description or question
I was investigating an issue that I was having (Sourcery was trying to move comment to out of the loop), when I found out a different issue that is more critical

```python
bla = {}

for i in range(45):
    if __import__("random").random() > 0.5:
        bla[i] = i

    other_var = max(bla.values())

print(other_var)
```
(I know that this is not an optimal code but this is what I fount to break when investigating)
Refactors to
```python
other_var = max(bla.values())  

bla = {i: i for i in range(45) if __import__("random").random() > 0.5}

print(other_var)
```
As you can see `other_variable` is defined before `bla` is defined

PS: I will try to find simplified case of the other issue and post it here

### Sourcery Version
0.8.0

### Code editor or IDE name and version

Discovered on NeoVim 0.5.0, but confirmed on VSCode

### OS name and version

Ubuntu 20.04 via WSL

**Comment by Hellebore** [comment]
Thanks for raising! Will get this addressed as soon as possible.

**Comment by Hellebore** [comment]
Fixed. Will be in the next release, which should be 0.8.2.

**Comment by Hellebore** [comment]
Fixed with release of 0.8.2.
